pkg/util/errors/errors.go

```diff
@@ -29,10 +29,16 @@ type Aggregate interface {
 // is itself an implementation of the error interface.  If the slice is empty,
 // this returns nil.
 func NewAggregate(errlist []error) Aggregate {
-	if len(errlist) == 0 {
+	errs := []error{}
+	for _, err := range errlist {
+		if err != nil {
```

Comment:
@thockin nil err is not the same as a nil pointer that implements the error interface: https://play.golang.org/p/7HPGL0z8SE (or maybe I misunderstand your first comment-- is there a typo?)

I don't like either choice of error aggregation-- better to define a type with an accumulate method that ignores nil errs, IMO. I agree w/ Tim that it's kinda weird to have errors and non-errors in the same list and I'd prefer it if we didn't encourage that.